### PR TITLE
[posix] logging enhancements to infra if

### PR DESCRIPTION
**Pull request by wgtdkp**
This PR includes two small enhancements to the posix infra if module:
1. simplify logging of socket errors since the `VerifyOrDie` macro already prints the line number and `strerror()`.
2. suppress "drop ICMPv6 message" debug logs when there is no input data to be read on the socket.

### CODE REVIEW

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

File: src/posix/platform/infra_if.cpp [hunk 1]
```diff
@@ -182,73 +178,41 @@ void platformInfraIfInit(otInstance *aInstance, const char *aIfName)
 
     // Initializes the infra interface.
     ifIndex = if_nametoindex(aIfName);
-    if (ifIndex == 0)
-    {
-        otLogCritPlat("failed to get the index for infra interface %s: %s", aIfName, strerror(errno));
-        DieNow(OT_EXIT_ERROR_ERRNO);
-    }
+    VerifyOrDie(ifIndex != 0, OT_EXIT_ERROR_ERRNO);
```
Comment (jwhui) [inline src/posix/platform/infra_if.cpp:181]: I think it is valuable to have more friendly error output for common user-facing issues (e.g. when specifying an interface that cannot be found). As an example, we added some more user-friendly error output in #5935.

Comment (simonlingoogle) [reply src/posix/platform/infra_if.cpp:181]: Agree with @jwhui . 

I think it's better to allow specifying an error message in `VerifyOrDie`. 

e.x.  
```cpp
VerifyOrDie(ifIndex != 0, OT_EXIT_ERROR_ERRNO, "failed to get the index for infra interface %s: %s", aIfName, strerror(errno));
```

Comment (wgtdkp) [reply src/posix/platform/infra_if.cpp]: Makes sense. reverted the change to the interface name check.

File: src/posix/platform/infra_if.cpp [hunk 2]
```diff
@@ -278,7 +246,7 @@ void platformInfraIfUpdateFdSet(fd_set &aReadFdSet, int &aMaxFd)
 
 void platformInfraIfProcess(otInstance *aInstance, const fd_set &aReadFdSet)
 {
-    otError  error = OT_ERROR_DROP;
+    otError  error = OT_ERROR_NONE;
```
Comment (librasungirl) [inline src/posix/platform/infra_if.cpp:249]: If update the initial value as in this PR,  below line# 318 (`error = OT_ERROR_NONE;` ) could be removed?
Just curious, any specific reason to change the initial value?

Comment (wgtdkp) [reply src/posix/platform/infra_if.cpp:249]: yes, line#318 can be removed, updated.
It is for suppress "drop ICMPv6 message" debug logs when there is no input data to be read on the socket.

**Review by librasungirl — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**